dashboard/src/app/workspaces/workspace-details/select-stack/recipe-authoring/workspace-recipe-authoring.controller.ts

```diff
@@ -109,8 +112,17 @@ export class WorkspaceRecipeAuthoringController {
 
   validateRecipe(content: string): void {
     this.recipeValidationError = '';
+
+    if (!content) {
+      return;
+    }
+
     try {
-      this.dockerfileParser.parse(content);
+      if (this.recipeFormatCopy === 'dockerfile') {
+        this.dockerfileParser.parse(content);
+      } else if (this.recipeFormatCopy === 'compose') {
+        this.composeParser.parse(content);
```

Comment:
Could we use constants instead those 'dockerfile' and 'compose' values or move this condition is separate service which will create needed instance?
WDYT?